wgengine/magicsock/magicsock.go

```diff
@@ -567,6 +567,108 @@ func (c *Conn) SetNetInfoCallback(fn func(*tailcfg.NetInfo)) {
 	}
 }
 
+// peerForIP returns the Node in that nm that's responsible for
+// handling the given IP address.
+func peerForIP(nm *controlclient.NetworkMap, ip netaddr.IP) (n *tailcfg.Node, ok bool) {
+	wgIP := wgcfg.IP{Addr: ip.As16()}
+	for _, p := range nm.Peers {
+		for _, a := range p.Addresses {
+			if a.IP == wgIP {
+				return p, true
+			}
+		}
+	}
+	for _, p := range nm.Peers {
+		for _, cidr := range p.AllowedIPs {
+			if cidr.Contains(wgIP) {
+				return p, true
+			}
+		}
+	}
+	return nil, false
+}
+
+// Ping handles a "tailscale ping" CLI query.
+func (c *Conn) Ping(ip netaddr.IP, cb func(*ipnstate.PingResult)) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	res := &ipnstate.PingResult{IP: ip.String()}
+	if c.privateKey.IsZero() {
+		res.Err = "local tailscaled stopped"
+		cb(res)
+		return
+	}
+	if c.netMap == nil {
+		res.Err = "no local netmap"
```

Comment:
If this happens and it gets logged at the top level as-is with log.Fatal, this'll be pretty incomprehensible to an end user. What can we do to decorate this? (Maybe the top log.Fatal, if appropriate, should say "internal error: %v", or the like?)

Reply:
I deleted this instead. The next "no matching peer" error can handle it instead.